src/components/course/course-header/CourseImportantDates.jsx

```diff
@@ -0,0 +1,112 @@
+import {
+  determineAllocatedCourseRunAssignmentsForCourse,
+  getSoonestEarliestPossibleExpirationData,
+  useCourseMetadata,
+  useRedeemablePolicies,
+} from '../../app/data';
+
+const messages = defineMessages({
+  importantDates: {
+    id: 'enterprise.course.about.page.important-dates.title',
+    defaultMessage: 'Important dates',
+    description: 'Title for the important dates section on the course about page',
+  },
+  enrollByDate: {
+    id: 'enterprise.course.about.page.important-dates.enroll-by-date',
+    defaultMessage: 'Enroll-by date',
+    description: 'Enroll-by date for the important dates section on the course about page',
+  },
+  courseStarts: {
+    id: 'enterprise.course.about.page.important-dates.course-starts',
+    defaultMessage: 'Course starts',
+    description: 'Course starts for the important dates section on the course about page in future tense',
+  },
+  courseStarted: {
+    id: 'enterprise.course.about.page.important-dates.course-started',
+    defaultMessage: 'Course started',
+    description: 'Course started the important dates section on the course about page in past tense',
+  },
+});
+
+const CourseImportantDates = () => {
+  const { courseKey } = useParams();
+  const { data: redeemableLearnerCreditPolicies } = useRedeemablePolicies();
+  const { data: courseMetadata } = useCourseMetadata();
+  const intl = useIntl();
+  const {
+    allocatedCourseRunAssignments,
+    allocatedCourseRunAssignmentKeys,
+    hasAssignedCourseRuns,
+  } = determineAllocatedCourseRunAssignmentsForCourse({
+    redeemableLearnerCreditPolicies,
+    courseKey,
+  });
+
+  const [searchParams] = useSearchParams();
+  const courseRunKey = searchParams.get('course_run_key')?.replaceAll(' ', '+');
+
+  if (!hasAssignedCourseRuns || (!!courseRunKey && !allocatedCourseRunAssignmentKeys.includes(courseRunKey))) {
+    return null;
+  }
+
+  const { date, soonestExpirationDateData } = getSoonestEarliestPossibleExpirationData({
+    assignmentObjectArray: allocatedCourseRunAssignments,
+    dateFormat: `${DATE_FORMAT} h:mm A`,
+  });
+  const soonestExpiringAllocatedAssignmentCourseStartDate = courseMetadata.availableCourseRuns.find(
+    (courseRun) => courseRun.key === soonestExpirationDateData?.contentKey,
+  )?.start;
+
+  const enrollByDate = date;
+  const courseStartDate = dayjs(soonestExpiringAllocatedAssignmentCourseStartDate).format(DATE_FORMAT);
+  const hasCourseStarted = dayjs(courseStartDate).isBefore(dayjs());
+
+  return (
+    <Stack gap={2} className="mt-4">
+      <h3>{intl.formatMessage(messages.importantDates)}</h3>
+      <section>
+        {enrollByDate && (
+          <Row>
+            <Col>
+              <Stack direction="horizontal" gap={2}>
+                <Icon src={Calendar} />
+                {intl.formatMessage(messages.enrollByDate)}
+              </Stack>
+            </Col>
+            <Col>
+              {enrollByDate}
+            </Col>
+          </Row>
```

Comment:
[suggestion] I think you could de-compose this further, by abstracting each date item into a sub-component, too. That way, the rendering here can focus solely on the i18n label and the date, without worrying about the layout as much.

```jsx
const CourseImportantDate = ({ label, children }) => (
  <Row className="course-important-date border-bottom mx-0 py-2">
    <Col className="px-0">
      <Stack direction="horizontal" gap={2}>
        <Icon src={Calendar} size="sm" />
        {label}
      </Stack>
    </Col>
    <Col>
      {children}
    </Col>
  </Row>
);
```

```jsx
{enrollByDate && (
  <CourseImportantDate label={intl.formatMessage(messages.enrollByDate)}>
    {enrollByDate}
  </CourseImportantDate>
)}
```